code/API_definitions/Optimal-Edge-Discovery.yaml

```diff
@@ -0,0 +1,741 @@
+        required: false
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Device"
+
+      responses:
+        "200":
+          description: returns optimal edge cloud zone matching query
+            parameters
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+          content:
+            application/json:
+              schema:
+                title: EdgeCloudZonesResponse
+                type: array
+                items:
+                  $ref: "#/components/schemas/ResourcesEdgeCloudZones"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "406":
+          $ref: "#/components/responses/Generic406"
+        "422":
+          $ref: "#/components/responses/Generic422"
+        "429":
+          $ref: "#/components/responses/Generic429"
+
+      tags:
+        - Discovery
+      summary: Discover optimal edge cloud zones for deployed applications
+      description: Returns a list of optimal edge cloud zones where you can
+         register your deployed application. You can choose to search without
+         passing any of the inputs parameters or a combination of Application
+         Profile and device information.
+
+components:
+  ######################################################
+  #  OAUTH
+  #######################################################
+  securitySchemes:
+    oAuth2:
+      type: oauth2
+      description: "This API uses OAuth 2 with the
+        implicit, password and client credentials grant flow"
+      flows:
+        implicit:
+          authorizationUrl: "https://example.com/oauth/authorize"
+          scopes:
+            "discovery:read": Grant read-only access to
+              discovery data
+        password:
+          tokenUrl: "https://example.com/oauth/token"
+          scopes:
+            "discovery:read": Grant read-only access to
+              discovery data
+        clientCredentials:
+          tokenUrl: "https://example.com/oauth/token"
+          scopes:
+            "discovery:read": Grant read-only access to
+              discovery data
+  ######################################################
+  #  RESOURCES
+  #######################################################
+  headers:
+    x-correlator:
+      description: |
+        When the API Consumer includes the "x-correlator" header in the request,
+        the API provider must include it in the response with the same value
+        that was used in the request. Otherwise, it is optional to include the
+        "x-correlator" header in the response with any valid value.
+        Recommendation is to use UUID for values.
+      required: false
+      schema:
+        type: string
```

Comment:
```suggestion
    x-correlator:
      description: |
        When the API Consumer includes the "x-correlator" header in the request,
         the API provider must include it in the response with the same value t
         hat was used in the request. Otherwise, it is optional to include the
         "x-correlator" header in the response with any valid value.
         Recommendation is to use UUID for values.
      required: false
      schema:
        type: string
        pattern: ^[a-zA-Z0-9-]{0,55}$
```